plugins/modem-manager/fu-mm-device.c

```diff
@@ -1477,7 +1504,17 @@ fu_mm_copy_firehose_prog(FuMmDevice *self, GBytes *prog, GError **error)
 	if (!fu_path_mkdir_parent(qcom_fw_dir, error))
 		return FALSE;
 
-	firehose_file_path = g_build_filename(qcom_fw_dir, "prog_firehose_sdx24.mbn", NULL);
+	if (self->device_id == 0x1001 || self->device_id == 0x1002)
+		firehose_file_path = g_build_filename(qcom_fw_dir, "prog_firehose_sdx24.mbn", NULL);
+	else if (self->device_id == 0x1004)
+		firehose_file_path = g_build_filename(qcom_fw_dir, "prog_firehose_sdx6x.elf", NULL);
+	else {
```

Comment:
Could we maybe use the quirk database for this? e.g.

    [MODEMMANAGER\FIREHOSE_DEVID_1001]
    ModemManagerFirehoseFilePath = prog_firehose_sdx24.mbn

and then do something like `fu_device_add_instance_u16(self, "FIREHOSE_DEVID", self->device_id)` and then  `fu_device_build_instance_id_quirk(self, NULL, "MODEMMANAGER", "FIREHOSE_DEVID", NULL);`

I think that'll scale much better for the future.